tests/ui/macros/stringify.rs

```diff
@@ -226,7 +226,7 @@ fn test_expr() {
     );
     c2_match_arm!(
         [ m! {} - 1 ],
-        "match () { _ => m! {} - 1, }",
+        "match () { _ => (m! {}) - 1, }", // parenthesis is redundant
```

Comment:
Can we correctly check to not add them?

Reply:
Done. I found an appealing way to solve this, requiring #124191.